src/soil/schemas/application.clj

```diff
@@ -1,17 +1,18 @@
 (ns soil.schemas.application
   (:require [schema.core :as s]))
 
-(s/defschema InterfaceType (s/enum :tcp :udp :http))
+(s/defschema InterfaceType (s/enum :tcp :udp :http :https :nrepl))
 (s/defschema ApplicationDefinition {:name       s/Str
                                     :devspace   s/Str
                                     :containers [{:name                       s/Str
                                                   :image                      s/Str
                                                   (s/optional-key :syncable?) s/Bool
                                                   :env                        (s/pred map?)}]
-                                    :interfaces [{:name      s/Str
-                                                  :port      s/Int
-                                                  :container s/Str
-                                                  :type      InterfaceType}]})
+                                    :interfaces [{:name                     s/Str
+                                                  :port                     s/Int
+                                                  (s/optional-key :expose?) s/Bool
```

Comment:
If you're going to add this, we probably want to change the adapter `application->ingress` to filter by this property

Reply:
https://github.com/formicarium/soil/pull/17/files#diff-52f04ec18cc774622c53aa0e9fbb2c54R102